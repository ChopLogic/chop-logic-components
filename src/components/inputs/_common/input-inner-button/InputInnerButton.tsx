import React from 'react';

import ChopLogicIcon, { CLIcon } from 'components/misc/icon/Icon';

import { StyledInputInnerButton } from './InputInnerButton.styled';

type ClearInputButtonProps = {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  label: string;
  icon: CLIcon;
};

const InputInnerButton: React.FC<ClearInputButtonProps> = ({ onClick, label, icon }) => {
  return (
    <StyledInputInnerButton onClick={onClick} aria-label={label} type='button'>
      <ChopLogicIcon name={icon} />
    </StyledInputInnerButton>
  );
};

export default InputInnerButton;
