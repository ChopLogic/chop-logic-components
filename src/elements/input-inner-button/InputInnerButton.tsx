import React from 'react';
import { ChopLogicIconName } from '@enums';
import styles from './InputInnerButton.module.scss';
import { ChopLogicIcon } from '@components/atoms';

type ClearInputButtonProps = {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  label: string;
  icon: ChopLogicIconName;
  disabled?: boolean;
};

const InputInnerButton: React.FC<ClearInputButtonProps> = ({ onClick, label, icon, disabled }) => {
  return (
    <button className={styles.inputButton} onClick={onClick} aria-label={label} type='button' disabled={disabled}>
      <ChopLogicIcon name={icon} />
    </button>
  );
};

export default InputInnerButton;
