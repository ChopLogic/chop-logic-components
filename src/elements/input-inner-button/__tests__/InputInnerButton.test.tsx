import { ChopLogicIconName } from '@enums';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';

import InputInnerButton from '../InputInnerButton';

import { DARK_THEME } from '@css';

describe('InputInnerButton', () => {
  const testProps = {
    onClick: vi.fn(),
    label: 'Test input',
    icon: ChopLogicIconName.Remove,
    theme: DARK_THEME,
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(<InputInnerButton {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have the accessible label', () => {
    render(<InputInnerButton {...testProps} />);
    expect(screen.getByLabelText(testProps.label)).toBeInTheDocument();
  });

  it('should call onClear handler', async () => {
    render(<InputInnerButton {...testProps} />);
    await userEvent.click(screen.getByRole('button'));
    expect(testProps.onClick).toHaveBeenCalledOnce();
  });

  it('should have type button', () => {
    render(<InputInnerButton {...testProps} />);
    expect(screen.getByRole('button')).toHaveAttribute('type', 'button');
  });
});
