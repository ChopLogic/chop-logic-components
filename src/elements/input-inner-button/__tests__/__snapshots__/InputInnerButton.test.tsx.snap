// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputInnerButton > should match the snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Test input"
    class="sc-egkSDF fydvLw"
    type="button"
  >
    <span
      aria-label="SVG Icon Fallback"
      style="font-size: 1rem;"
    >
      â¬œ
    </span>
  </button>
</DocumentFragment>
`;
